tests/test_welltest_dpds.py

```diff
@@ -0,0 +1,164 @@
+import os
+import sys
+import subprocess
+import pytest
+import numpy as np
+from ecl.summary import EclSum
+from subscript.welltest_dpds import welltest_dpds
+
+ECLDIR = os.path.join(os.path.dirname(__file__), "data/welltest/eclipse/model")
+ECLCASE = "DROGON_DST_PLT-0"
+
+
+@pytest.mark.integration
+def test_integration():
+    """Test that endpoint is installed"""
+    assert subprocess.check_output(["welltest_dpds", "-h"])
+
+
+@pytest.mark.integration
+def test_main(tmpdir):
+    """Test invocation from command line"""
+    tmpdir.chdir()
+
+    datafilepath = os.path.join(ECLDIR, ECLCASE)
+
+    # defaults only
+    sys.argv = ["welltest_dpds", datafilepath, "55_33-1"]
+    welltest_dpds.main()
+    assert os.path.exists("wbhp.csv")
+    os.unlink("wbhp.csv")
+
+    # test --outfilessufix
+    sys.argv = [
+        "welltest_dpds",
+        datafilepath,
+        "55_33-1",
+        "--outfilessufix",
+        "blabla",
+    ]
+    welltest_dpds.main()
+    assert os.path.exists("wbhp_blabla.csv")
+    os.unlink("wbhp_blabla.csv")
+
+    # test --outputdirectory
+    sys.argv = [
+        "welltest_dpds",
+        datafilepath,
+        "55_33-1",
+        "--outputdirectory",
+        "blabla",
+    ]
+    with pytest.raises(FileNotFoundError, match=r".*No such outputdirectory.*"):
+        welltest_dpds.main()
+
+    os.mkdir("blabla")
+    welltest_dpds.main()
+    assert os.path.exists("./blabla/wbhp.csv")
+    os.unlink("blabla/wbhp.csv")
+
+    # test --pahse
+    sys.argv = ["welltest_dpds", datafilepath, "55_33-1", "--phase", "GAS"]
+    welltest_dpds.main()
+    assert os.path.exists("wgpr.csv")
+
+
+def test_get_summary_vec():
+    """Test that summary reading is handled correctly"""
+
+    datafilepath = os.path.join(ECLDIR, ECLCASE)
+    summary = EclSum(datafilepath)
+    with pytest.raises(KeyError, match=r".*No such key.*"):
+        welltest_dpds.get_summary_vec(summary, "no_well")
+        welltest_dpds.get_summary_vec(summary, "NOVEC:55_33-1")
+
+    wopr = welltest_dpds.get_summary_vec(summary, "WOPR:55_33-1")
+    assert len(wopr) == 556
+
+
+def test_get_buildup_indices():
+    """Test that buildup periods are identified correct """
+
+    datafilepath = os.path.join(ECLDIR, ECLCASE)
+    summary = EclSum(datafilepath)
+
+    wbhp = welltest_dpds.get_summary_vec(summary, "WBHP:55_33-1")
+    bu_start, bu_end = welltest_dpds.get_buildup_indices(wbhp)
+    assert bu_start == []
+    assert bu_end == []
+
+    wopr = welltest_dpds.get_summary_vec(summary, "WOPR:55_33-1")
+    bu_start, bu_end = welltest_dpds.get_buildup_indices(wopr)
+    assert bu_start == [7, 260]
+    assert bu_end == [254, 555]
+
+
+def test_get_supertime():
```

Comment:
We should have a couple of additional tests with explicit arrays as input, not requiring a summary file to test the function. Starting with the empty array, and then increasing number of elements. pytest.mark.parametrize is a good fit.